### ISSUE TITLE
Provide a method of caching for the large wheels. 

### ISSUE THREAD

**Issue by jfmherokiller** [oobabooga/text-generation-webui#6591]
**Description**

Setup a download setup (if possible) wherein the big wheels:
llama_cpp_python_cuda_tensorcores
exllamav2
llama_cpp_python_cuda
are downloaded once and then hash checked locally instead of being redownloaded every single time. 

This should be done because these combined wheels are about 1GB and fail to be cached by pip. This is seemingly due to them being hosted by GitHub. 

I will admit this seems to be a general issue with pip as seen here https://discuss.python.org/t/what-are-the-caching-rules-for-wheels-installed-from-urls/21594/2

**Additional Context**

If applicable, please provide any extra information, external links, or screenshots that could be useful.

**Comment by jfmherokiller** [comment]
while also admit this may be slightly streching the application design goal. Due to the fact that this application seems to want to be an everything ai gen adding some accomations for those who dont have unstable internet. Because I dread any time i need to update this application due to the download speeds.

**Comment by oobabooga** [comment]
This happens with every Python project that has a `.whl` in the requirements. I think that the only solution is to publish a package to pypi and put that in the requirements instead of the whl. It doesn't seem to be possible to solve it through flags, as per my research.

**Comment by jfmherokiller** [comment]
yep the only way i saw a possible version was a local simpleindex that would be generated and kept up to date nearby.
